pkg/ddevapp/config.go

```diff
@@ -243,22 +315,28 @@ func (c *Config) Hostname() string {
 
 // WriteDockerComposeConfig writes a docker-compose.yaml to the app configuration directory.
 func (c *Config) WriteDockerComposeConfig() error {
-	log.WithFields(log.Fields{
-		"Location": c.DockerComposeYAMLPath(),
-	}).Debug("Writing docker-compose.yaml")
+	var err error
 
-	f, err := os.Create(c.DockerComposeYAMLPath())
-	if err != nil {
-		return err
-	}
-	defer util.CheckClose(f)
+	if !fileutil.FileExists(c.DockerComposeYAMLPath()) {
+		log.WithFields(log.Fields{
+			"Location": c.DockerComposeYAMLPath(),
+		}).Debug("Writing docker-compose.yaml")
 
-	rendered, err := c.RenderComposeYAML()
-	if err != nil {
-		return err
-	}
-	_, err = f.WriteString(rendered)
+		f, err := os.Create(c.DockerComposeYAMLPath())
+		if err != nil {
+			return err
+		}
+		defer util.CheckClose(f)
 
+		rendered, err := c.RenderComposeYAML()
+		if err != nil {
+			return err
+		}
+		_, err = f.WriteString(rendered)
+		if err != nil {
+			return err
+		}
+	}
 	return err
```

Comment:
If I'm not mistaken, this would be clearer by removing the declaration of err at the top and then returning nil here. Because err is not going to be an error right? (and err is shadowed in every stanza above)